src/extension/background-script/actions/allowances/enable.ts

```diff
@@ -1,17 +1,17 @@
-import { Runtime } from "webextension-polyfill";
 import utils from "~/common/lib/utils";
 import db from "~/extension/background-script/db";
-import type { MessageAllowanceEnable } from "~/types";
+import type { MessageAllowanceEnable, Sender } from "~/types";
 
 import state from "../../state";
-import { setIcon, ExtensionIcon } from "../setup/setIcon";
+import { ExtensionIcon, setIcon } from "../setup/setIcon";
+
+const enable = async (message: MessageAllowanceEnable, sender: Sender) => {
+  let host;
```

Comment:
Could you extract these checks to a function? It’s duplicated in many files

Reply:
👍 put it in a helper. 

and add a note like the one from ublock: 

// https://github.com/uBlockOrigin/uBlock-issues/issues/1992
//   If present, use MessageSender.origin to determine whether the port is
//   from a privileged page, otherwise use MessageSender.url.
//   MessageSender.origin is more reliable as it is not spoofable by a
//   compromised renderer.